test-framework/src/main/java/org/apache/brooklyn/test/framework/TargetableTestComponentImpl.java

```diff
@@ -116,4 +125,14 @@ private static Entity getTargetById(final ExecutionContext executionContext, fin
     protected <T> T getRequiredConfig(ConfigKey<T> key) {
         return checkNotNull(config().get(key), "config %s must not be null", key);
     }
+
+    protected void setUpAndRunState(boolean up, Lifecycle status) {
+        if (up) {
+            sensors().set(SERVICE_UP, up);
+            setExpectedState(this, status);
+        } else {
+            setExpectedState(this, status);
+            sensors().set(SERVICE_UP, up);
+        }
```

Comment:
Agreed we should change that. Will it mean that app.start() can return when the service.isUp still equals false? I believe so - e.g. the `TestCase` just calls start sequentially on its children, without waiting for service.isUp.

If so, we'll also need to change our test assertions to "eventually".

I'll hold off doing that until another PR.